### PR TITLE
Moonlander: Add RGB LED layout map macro

### PR DESCRIPTION
**Pull request by joukewitteveen**
## Description

A macro that allows pretty definitions of RGB LED maps. The macro name `LED_LAYOUT_moonlander` is rather generic, maybe `RGB_LAYOUT_moonlander` was better, but it is in line with the macro name used for _ergodox_ez_ (`LED_LAYOUT_ergodox_pretty`) and the name used for the keyhive _sofle_ (`LED_LAYOUT`).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by joukewitteveen** [comment]
This PR is the result of a [piece of documentation](https://docs.qmk.fm/#/contributing?id=keymaps):
> Do not lump features in with keymap PR’s. Submit the feature first and then a second PR for the keymap.

In other words, I would like to use the macro proposed here in a keymap that I intend to open a merge request for.

**Review by drashna — APPROVED** [review]

File: keyboards/moonlander/moonlander.c [hunk 1]
```diff
@@ -276,39 +276,21 @@ led_config_t g_led_config = { {
...
-    1, 4, 4, 4, 4, 4,
-    4, 4, 1, 1, 1, 1,
-    1, 1, 1, 1, 1, 4,
-    4, 4, 4, 1, 4, 4,
-    4, 4, 1, 4, 4, 4,
-    4, 1, 4, 4, 4, 4,
-    1, 4, 4, 4, 4, 4,
-    4, 4, 1, 1, 1, 1
-} };
+}, LED_LAYOUT_moonlander(
```
Comment (joukewitteveen) [inline keyboards/moonlander/moonlander.c:279]: Unfortunately, `qmk lint` (`lib/python/qmk/c_parse.py`) does not like the use of macros here. Reverting would be easy, but using this macro brought to light some issues, so there is value in using a macro here.

File: keyboards/moonlander/moonlander.c [hunk 2]
```diff
@@ -276,39 +276,21 @@ led_config_t g_led_config = { {
...
+    {   0,   0 }, {  17,   0 }, {  34,   0 }, {  51,   0 }, {  68,   0 }, {  86,   0 }, { 105,   0 },    { 146,   0 }, { 163,   0 }, { 181,   0 }, { 198,   0 }, { 216,   0 }, { 233,   0 }, { 250,   0 },
```
Comment (joukewitteveen) [inline keyboards/moonlander/moonlander.c:280]: I noted that the steps in the x-coordinate are uneven. They are mostly 17, but between 86 and 105 is a step size of 19. Also, the distribution of some excess horizontal spacing is uneven: the columns between which we have a step size of 18 differ between the left and right halves.

Moreover, the midpoint (set to 125) is not halfway between 105 and 146.

File: keyboards/moonlander/moonlander.c [hunk 3]
```diff
@@ -276,39 +276,21 @@ led_config_t g_led_config = { {
...
+    {   0,  12 }, {  17,  12 }, {  34,  12 }, {  51,  12 }, {  68,  12 }, {  86,  12 }, { 105,  12 },    { 146,  12 }, { 163,  12 }, { 181,  12 }, { 198,  12 }, { 216,  12 }, { 233,  12 }, { 250,  12 },
+    {   0,  25 }, {  17,  25 }, {  34,  25 }, {  51,  25 }, {  68,  25 }, {  86,  25 }, { 105,  25 },    { 146,  25 }, { 163,  25 }, { 181,  25 }, { 198,  25 }, { 216,  25 }, { 233,  25 }, { 250,  25 },
+    {   0,  38 }, {  17,  38 }, {  34,  38 }, {  51,  38 }, {  68,  38 }, {  86,  38 }, { 105,  38 },    { 146,  38 }, { 163,  38 }, { 181,  38 }, { 198,  38 }, { 216,  38 }, { 233,  38 }, { 250,  38 },
+    {   0,  51 }, {  17,  51 }, {  34,  51 }, {  51,  51 }, {  68,  51 },               { 116,  59 },    { 131,  59 },               { 181,  51 }, { 198,  51 }, { 216,  51 }, { 233,  51 }, { 250,  51 },
+                                                            {  90,  55 }, { 105,  68 }, { 116,  86 },    { 146,  86 }, { 161,  68 }, { 161,  55 }
```
Comment (joukewitteveen) [inline keyboards/moonlander/moonlander.c:285]: Something weird is going on with the column coordinates of the right thumb cluster. The coordinates in the left thumb cluster seem reasonable.

Before addressing this or the other layout issue, I would like to know how these coordinates were obtained. @drashna, do you know?

**Review by joukewitteveen — COMMENTED** [review]
I added usage of the led layout macro to the keyboard source and discover some oddities in the coordinate map.

File: keyboards/moonlander/moonlander.c (hunk 1 above)
Comment (fauxpark) [reply keyboards/moonlander/moonlander.c:279]: See this thread https://github.com/qmk/qmk_firmware/pull/18973#discussion_r1018571012

Comment (joukewitteveen) [reply keyboards/moonlander/moonlander.c:279]: I don't believe `g_led_config` can be represented in `info.json` yet, can it? I guess the benefits provided by using a macro here (i.e. catching mapping oddities) are out of reach to the DD approach, or are there some cool plans in this direction?

Comment (fauxpark) [reply keyboards/moonlander/moonlander.c:279]: It's possible, but a lot of the data-driven stuff is not well documented just yet, and some things are only in develop (which should merge in a day or so).
https://github.com/qmk/qmk_firmware/blob/master/keyboards/1upkeyboards/pi60_hse/info.json#L31-L51
https://github.com/qmk/qmk_firmware/blob/master/data/schemas/keyboard.jsonschema#L294-L320

Comment (joukewitteveen) [reply keyboards/moonlander/moonlander.c:279]: Hmm, unfortunately the order in the json file is the LED indexing order. This precludes generating a layout macro from them automatically. Automatic generation on the basis of a key layout _is_ possible, but only if every led has a position on the matrix that also has a key assigned to it. Even then, it is quite involved and we don't have a way to configure the name of the macro (`LED_{layout_name}` is possible, though).

I guess for now this leaves no other option than to set the LED layout macro the old fashioned way in `moonlander.h`, but revert its usage in defining `g_led_config`. I'll do that once I know what to do with the weird positions currently set in `g_led_config`.

**Review by tzarc — APPROVED** [review]

**Pull request merged**